battleship.py

```diff
@@ -131,15 +137,19 @@ def addShips(grid, numShips):
 Returns: None
 '''
 def drawGrid(data, canvas, grid, showShips):
+    cell =data["cellsize"]
     for row in range(data["rows"]):
-     for col in range(data["cols"]): 
-      if grid[row][col]== SHIP_UNCLICKED:
-        canvas.create_rectangle(data["cellsize"]*row, data["cellsize"]*col, data["cellsize"]*(row+1), data["cellsize"]*(col+1), fill="yellow")
-      else:
-        canvas.create_rectangle(data["cellsize"]*row, data["cellsize"]*col, data["cellsize"]*(row+1), data["cellsize"]*(col+1), fill= "blue")
-    return 
- 
-
+        for col in range(data["cols"]): 
+            if grid[row][col]== SHIP_UNCLICKED:
+                canvas.create_rectangle(cell*col, cell*row, cell*(col+1), cell*(row+1), fill="yellow")
+            elif grid[row][col]==  EMPTY_UNCLICKED:
+                canvas.create_rectangle(cell*col, cell*row, cell*(col+1), cell*(row+1), fill= "blue")
+            elif  grid[row][col]== SHIP_CLICKED:
+                canvas.create_rectangle(cell*col, cell*row, cell*(col+1), cell*(row+1), fill="red")
+            elif grid[row][col]== EMPTY_CLICKED:
+                canvas.create_rectangle(cell*col, cell*row, cell*(col+1), cell*(row+1), fill="white")
+            if grid[row][col]==  SHIP_UNCLICKED and showShips== False:
+                canvas.create_rectangle(cell*col, cell*row, cell*(col+1), cell*(row+1), fill= "blue")
```

Comment:
You can reorder the if conditions so that you don't need to write a separate if for when showships is false